### PR TITLE
k8sreporter test

### PR DESCRIPTION
**Pull request by RemindD**
as title to improve test coverage

### CODE REVIEW

File: coa/pkg/apis/v1alpha2/providers/reporter/k8s/k8sreporter_test.go [hunk 1]
```diff
@@ -38,3 +47,96 @@ func TestGet(t *testing.T) {
 	}, false)
 	assert.Nil(t, err)
 }
+
+func TestInitWithMap(t *testing.T) {
```
Comment (Haishi2016) [inline coa/pkg/apis/v1alpha2/providers/reporter/k8s/k8sreporter_test.go:51]: Update the test name to a more appropriate name.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -76,7 +76,12 @@ jobs:
         cd ..
 
     - name: COA Test
-      run: cd coa && mage cleanTest
+      run: |
```
Comment (msftcoderdjw) [inline .github/workflows/go.yml:79]: Do need to do like this, you can just mv go.work.bk to go.work. https://github.com/eclipse-symphony/symphony/blob/034b25fc13f9de80a6345dfcd724d683eb69876a/.github/workflows/integration.yml#L83C1-L85C30

More importantly, I double you have some missing dependencies in go.mod in COA. When running unit test, usually it is not needed to add go.work.

File: coa/pkg/apis/v1alpha2/providers/reporter/k8s/k8sreporter_test.go (hunk 1 above)
Comment (RemindD) [reply coa/pkg/apis/v1alpha2/providers/reporter/k8s/k8sreporter_test.go]: Fixed

File: .github/workflows/go.yml (hunk 2 above)
Comment (RemindD) [reply .github/workflows/go.yml]: I have updated go.mod in COA

**Review by Haishi2016 — APPROVED** [review]

**Pull request merged**